hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java

```diff
@@ -72,6 +78,14 @@ public void enrichTrace(StructuredTrace trace) {
         .forEach(pair -> decorateWithBackendEntity(pair.getRight().get(), pair.getLeft(), trace));
   }
 
+  @Override
+  public void enrichEvent(StructuredTrace trace, Event event) {
+    String backendOperation = SpanAttributeUtils.getFirstAvailableStringAttribute(event, BACKEND_OPERATIONS);
```

Comment:
You have to move this code as a function in `DbSemanticConventionUtils`, and used it in all the backendResolver except `http`, `grpc` and `client`

In `httpBacendResolver` -> operation will be value of http.schema
In `grpcBackendResolver` -> operation will be `grpc.request.method`

The rest of the resolver will have from `BACKEND_OPERATIONS` as above.

Reply:
Again, depends on if we want to add the `db.operation` in backend entity as an attribute. If yes,

> In httpBacendResolver -> operation will be value of http.schema

I believe that's already taken care here https://github.com/hypertrace/hypertrace-ingester/blob/main/hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/HttpBackendResolver.java#L50

> In grpcBackendResolver -> operation will be grpc.request.method

I believe that's already take care here https://github.com/hypertrace/hypertrace-ingester/blob/main/hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/GrpcBackendResolver.java#L36

Reply:
But, that value should be set in the `BACKEND_OPERATON` attribute.